### PR TITLE
Also skip JAVA_TOOL_OPTIONS on Windows

### PR DESCRIPTION
**Pull request by jasontedor**
On non-Windows platforms, we ignore the environment variable JAVA_TOOL_OPTIONS (this is an environment variable that the JVM respects by default for picking up extra JVM options). The primary reason that we ignore this because of the Jayatana agent on Ubuntu; a secondary reason is that it produces an annoying "Picked up JAVA_TOOL_OPTIONS: ..." output message. When the elasticsearch-env batch script was introduced for Windows, ignoring this environment variable was deliberately not carried over as the primary reason does not apply on Windows. However, after additional thinking, it seems that we should simply be consistent to the extent possible here (and also avoid that annoying "Picked up JAVA_TOOL_OPTIONS: ..." on Windows too). This commit causes the Windows version of elasticsearch-env to also ignore JAVA_TOOL_OPTIONS.

### CODE REVIEW

File: distribution/src/main/resources/bin/elasticsearch-env.bat [hunk 1]
```diff
@@ -28,6 +28,13 @@ if not exist %JAVA% (
   exit /B 1
 )
 
+rem don't let JAVA_TOOL_OPTIONS slip in (e.g., agents in Ubuntu so this works
```
Comment (tlrx) [inline distribution/src/main/resources/bin/elasticsearch-env.bat]: I don't think we should talk about Ubuntu here

File: distribution/src/main/resources/bin/elasticsearch-env.bat [hunk 2]
```diff
@@ -28,6 +28,13 @@ if not exist %JAVA% (
...
+rem around https://bugs.launchpad.net/ubuntu/+source/jayatana/+bug/1441487)
+if not "%JAVA_TOOL_OPTIONS%" == "" (
```
Comment (tlrx) [inline distribution/src/main/resources/bin/elasticsearch-env.bat]: I'd be more comfortable if it was documented in the "Install Elasticsearch with zip on Windows". We have few notes there already.

**Review by tlrx — CHANGES_REQUESTED** [review]
I left two comments

File: distribution/src/main/resources/bin/elasticsearch-env.bat (hunk 1 above)
Comment (jasontedor) [inline distribution/src/main/resources/bin/elasticsearch-env.bat]: Okay, I removed it.

File: distribution/src/main/resources/bin/elasticsearch-env.bat (hunk 2 above)
Comment (jasontedor) [inline distribution/src/main/resources/bin/elasticsearch-env.bat]: This is a good suggestion, I pushed more general docs.

**Comment by jasontedor** [comment]
@tlrx I addressed your feedback, would you mind taking another look?

**Review by tlrx — APPROVED** [review]
LGTM, thanks @jasontedor

**Comment by jasontedor** [comment]
Thank you @tlrx.

**Pull request merged**